R/dataRetrieval.R

```diff
@@ -0,0 +1,308 @@
+#' 
+#' The dataset entails 358 mouse RNA-seq samples annotated to 18 main cell types: 
+#' Adipocytes, Astrocytes, B cells, Cardiomyocytes, Dendritic cells, Endothelial 
+#' cells, Epithelial cells, Erythrocytes, Fibroblasts, Granulocytes, Hepatocytes,
+#' Macrophages, Microglia, Monocytes, Neurons, NK cells, Oligodendrocytes, T cells.
+#' 
+#' @return A \linkS4class{SummarizedExperiment} object where \code{assay(..., "normcounts")}
+#' contains the normalized expression values and \code{colData()} can be used to 
+#' extract the cell type labels of every sample.
+#'
+#' @author Friederike Duendar
+#' 
+#' @references
+#' Benayoun B et al. (2019).
+#' Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses.
+#' \emph{Genome Research}. doi: 10.1101/gr.240093.118
+#' 
+#' @examples
+#' ref.se <- MouseBulkData()
+#' 
+#' \dontrun{
+#' ## use the reference data to predict labels for the cells of your scRNA-seq data
+#' predicted_labels <- SingleR(test = your_SCE,
+#'                            training = assay(ref.se, "normcounts"), 
+#'                            labels = colData(ref.se)$label.main)
+#' }                  
+#' @export
+#' 
+MouseBulkData <- function(){
+    version <- "1.0.0"
+    se <- .create_se(file.path("mouse.rnaseq", version), 
+                     assays="normcounts", rm.NA = NULL,
+                     has.rowdata = FALSE, has.coldata = TRUE)
+}
+
+
+#####################################################################
+### Helper function
+#####################################################################
+
+#' Create a SummarizedExperiment object using data from ExperimentHub
+#' 
+#' @param dataset string indicating the name of the reference data set to be
+#' retrieved. Choices are: c("hpca", "blueprint_encode", "immgen", "mouse.rnaseq")
+#' @param rm.NA indicate how to handle NA's. Valid choices are "rows" (remove
+#' genes with at least one missing value), "cols" (remove samples with at least
+#' one missing value), "both" (remove samples as well as genes that have at least
+#' one missing value, respectively), "none" or NULL.
+#' 
+#' @importFrom ExperimentHub ExperimentHub
+#' @importFrom SummarizedExperiment SummarizedExperiment
+#' @importFrom SummarizedExperiment rowData
+.create_se <- function(dataset = c("hpca", "blueprint_encode", "immgen", "mouse.rnaseq"),
+                       hub = ExperimentHub(), assays="normcounts",
+                       rm.NA = c("rows","cols","both","none"),
+                       has.rowdata=FALSE, has.coldata=TRUE) {
+    
+    ## TEMPORARY CODE for pulling data directly from github until it is 
+    ## available on ExHub =====================================================
+    ref.set <- getReferenceDataset(dataset=dataset)
+    nrmcnts <- ref.set$data
+    colnames(nrmcnts) <- paste(colnames(nrmcnts), 1:ncol(nrmcnts), sep = ".")
+    
+    ## handling NA's
+    if(!is.null(rm.NA)){
```

Comment:
Why not allow this to be handled by `"none"`? There's no need for special `NULL` behavior.

Reply:
personal indulgence.